network/p2p/libp2pNode.go

```diff
@@ -19,57 +19,24 @@ import (
-	// it will be notified when a new disallow list update is distributed.
-	DisallowListNotificationConsumer
-	// CollectionClusterChangesConsumer  is the interface for consuming the events of changes in the collection cluster.
-	// This is used to notify the node of changes in the collection cluster.
-	// LibP2PNode implements this interface and consumes the events to be notified of changes in the clustering channels.
-	// The clustering channels are used by the collection nodes of a cluster to communicate with each other.
-	// As the cluster (and hence their cluster channels) of collection nodes changes over time (per epoch) the node needs to be notified of these changes.
-	CollectionClusterChangesConsumer
-	// DisallowListOracle exposes the disallow list oracle API for external consumers to query about the disallow list.
-	DisallowListOracle
+// Node single node management capabilities
+type Node interface {
 	// Start the libp2p node.
 	Start(ctx irrecoverable.SignalerContext)
 	// Stop terminates the libp2p node.
 	Stop() error
-	// AddPeer adds a peer to this node by adding it to this node's peerstore and connecting to it.
-	AddPeer(ctx context.Context, peerInfo peer.AddrInfo) error
-	// RemovePeer closes the connection with the peer.
-	RemovePeer(peerID peer.ID) error
-	// GetPeersForProtocol returns slice peer IDs for the specified protocol ID.
-	GetPeersForProtocol(pid protocol.ID) peer.IDSlice
-	// CreateStream returns an existing stream connected to the peer if it exists, or creates a new stream with it.
-	CreateStream(ctx context.Context, peerID peer.ID) (libp2pnet.Stream, error)
 	// GetIPPort returns the IP and Port the libp2p node is listening on.
 	GetIPPort() (string, string, error)
-	// RoutingTable returns the node routing table
-	RoutingTable() *kbucket.RoutingTable
-	// ListPeers returns list of peer IDs for peers subscribed to the topic.
-	ListPeers(topic string) []peer.ID
-	// Subscribe subscribes the node to the given topic and returns the subscription
-	Subscribe(topic channels.Topic, topicValidator TopicValidatorFunc) (Subscription, error)
-	// UnSubscribe cancels the subscriber and closes the topic.
-	UnSubscribe(topic channels.Topic) error
-	// Publish publishes the given payload on the topic.
-	Publish(ctx context.Context, topic channels.Topic, data []byte) error
 	// Host returns pointer to host object of node.
 	Host() host.Host
+}
+
+// PeerManagement set of node traits related to its lifecycle and metadata retrieval
+type PeerManagement interface {
+	// SetComponentManager sets the component manager for the node.
+	// SetComponentManager may be called at most once.
+	SetComponentManager(cm *component.ComponentManager)
+
 	// WithDefaultUnicastProtocol overrides the default handler of the unicast manager and registers all preferred protocols.
 	WithDefaultUnicastProtocol(defaultHandler libp2pnet.StreamHandler, preferred []protocols.ProtocolName) error
```

Comment:
`WithDefaultUnicastProtocol` does not belong to the `PeerManagement` domain, it is more relevant to `CreateStream` method.

Reply:
https://github.com/onflow/flow-go/pull/4630/commits/455147a08f62bace4ff3fb18c2cc6ae018d3ca68